torch/ao/nn/sparse/quantized/linear.py

```diff
@@ -157,26 +159,24 @@ def set_weight_bias(self, w: torch.Tensor, b: Optional[torch.Tensor],
         self._packed_params.set_weight_bias(w, b, row_block_size, col_block_size)
 
     @classmethod
-    def from_float(cls, mod):
+    def from_float(cls, mod, row_block_size=1, col_block_size=4):
         r"""Create a quantized sparse module from a float module.
 
         We only care about the convert at this stage, no need for observers just yet.
+
+        TODO: Need to figure out how to store the block shapes in the mod
         """
-        assert type(mod) == cls._FLOAT_MODULE, ' nnq.' + cls.__name__ + '.from_float only works for ' + \
-            cls._FLOAT_MODULE.__name__
+        assert type(mod) in cls._FLOAT_MODULE, cls._get_name() + \
+            '.from_float only works for ' + \
+            str([n.__name__ for n in cls._FLOAT_MODULE])
         # TODO: Need to add options to qconfig to avoid the calibration.
         # TODO: Add calibration for the sparsity
         assert hasattr(mod, 'qconfig'), 'Input float module must have qconfig defined'
         activation_post_process = mod.activation_post_process
-        if type(mod) == nni.LinearReLU:
-            mod = mod[0]
         weight_post_process = mod.qconfig.weight()
 
         # It is important to multiply by the mask BEFORE calling the `weight_post_process`
-        # TODO (zaf): Mask might not be part of the qconfig (T83295194)
-        weight = mod.weight
-        if getattr(mod.qconfig, 'mask', False):
-            weight = mod.qconfig.mask * mod.weight
+        weight = mod.weight * mod.mask
```

Comment:
Let us make mask an attribute of the fakeSparse Module and do this via reparameterization